src/tests/jobtest.c

```diff
@@ -30,26 +30,131 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <strings.h>
 #include <daos.h>
+#include <daos_mgmt.h>
+#include <daos/common.h>
+#include <daos/pool.h>
+
+static char *progname;
+
+void print_usage(void)
+{
+	fprintf(stderr, "Usage: %s -p pool_str [-s nsecs] [-x] "
+			"[-h handles_per_pool]\n", progname);
+}
+
+void cleanup_handles(uuid_t *pool_uuids, int num_pools,
+		     daos_handle_t **pool_handles, int handles_per_pool)
+{
+	int i, rc;
+
+	for (i = 0; i < num_pools; i++) {
+		int idx;
+
+		if (uuid_is_null(pool_uuids[i])) {
+			continue;
+		}
+		for (idx = 0; idx < handles_per_pool; idx++) {
+			if (daos_handle_is_inval(pool_handles[i][idx])) {
+				continue;
+			}
+			rc = daos_pool_disconnect(pool_handles[i][idx], NULL);
+			if (rc) {
+				struct dc_pool	*pool;
+				char		pool_str[64];
+				char		hdl_str[64];
+
+				pool = dc_hdl2pool(pool_handles[i][idx]);
+				uuid_unparse_lower(pool->dp_pool, pool_str);
+				uuid_unparse_lower(pool->dp_pool, hdl_str);
+				printf("disconnect handle %s from pool %s "
+					"failed\n", hdl_str, pool_str);
+			}
+		}
+		free(pool_handles[i]);
+		pool_handles[i] = NULL;
+	}
+}
+
+/*
+ * Takes in a comma separated list of UUIDs and turns them into a list of
+ * UUID structures. It will return the number of entries on success and -1 on
+ * failure
+ */
+int parse_pool_handles(char *hndl_str, uuid_t **handles)
+{
+	char		*idx, *pool_str;
+	const char	delim[2] = ",";
+	int		hdl_count = 1 /* we have at least 1 */, i = 0, rc;
+	uuid_t		*hndls = NULL;
+
+	/* Figure out how many handles are in the string */
+	idx = strchr(hndl_str, ',');
+	while (idx != NULL) {
+		hdl_count++;
+		idx = strchr(idx + 1, ',');
+	}
+
+	hndls = calloc(hdl_count, sizeof(uuid_t));
+	if (!hndls) {
+		perror("Unable to allocate space for pool handle list");
+		return -1;
+	}
+
+	/* Parse the UUIDS into their structures */
+	pool_str = strtok(hndl_str, delim);
```

Comment:
Does this function need to be re-entrant safe?  If so, then use strtok_r()

Reply:
It shouldn't need to be re-entrant. There is only one thread of execution in this program at this point. The daos library will make ULTs but at this point we aren't using the DAOS library.